File: LeetCode_Python/length_of_last_word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        final_word = []

        for i in range(len(s) - 1, -1, -1):
            if s[i] != " " : final_word.append(s[i])
            else:
                if len(final_word) != 0 : break

        return len(final_word)
    
s = "Hello World"
sol = Solution()
print(sol.lengthOfLastWord(s))

File: LeetCode_C/length_of_last_word.c
#include <stdio.h>
#include <string.h>

int lengthOfLastWord(char* s) {
    int s_len = strlen(s);
    int length = 0;
    
    for (int i = s_len - 1; i >= 0; i--) {
        if (s[i] != ' ') {
            length ++;
        } else {
            if (length != 0) {
                break;
            }
        }
    }
    return length;
}

int main() {
    char s[] = "Hello World";
    int ans = lengthOfLastWord(s);
    printf("%d", ans);
    return 0;
}